### PR TITLE
Fix ImageToHistogramFilter with AutoMinumumMaximuim

### PR DESCRIPTION
**Pull request by blowekamp**
## PR Checklist

- [X] :no_entry_sign: [Makes breaking changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes)
- [X] :no_entry_sign: [Makes design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes)
- [X] :no_entry_sign: Adds the License notice to new files.
- [X] :no_entry_sign: Adds Python wrapping.
- [X] :no_entry_sign: Adds tests and baseline comparison (quantitative).
- [X] :no_entry_sign: [Adds test data](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Documentation/UploadBinaryData.md).
- [X] :no_entry_sign: Adds Examples to [ITKExamples](https://github.com/InsightSoftwareConsortium/ITKExamples)
- [X] :no_entry_sign: Adds Documentation.

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Comment by blowekamp** [comment]
@lassoan This patch contains the pipeline fix needed for the vtkImageThresholdCalculator failure you encountered.

File: Modules/Numerics/Statistics/test/itkImageToHistogramFilterTest2.cxx [hunk 1]
```diff
@@ -27,11 +27,18 @@ itkImageToHistogramFilterTest2(int argc, char * argv[])
   if (argc < 3)
   {
     std::cerr << "Missing command line arguments" << std::endl;
-    std::cerr << "Usage :  " << argv[0] << " inputRGBImageFileName outputHistogramFile.txt" << std::endl;
+    std::cerr << "Usage :  " << argv[0] << " inputRGBImageFileName outputHistogramFile.txt (autoMinumumMaximum)"
```
Comment (dzenanz) [inline Modules/Numerics/Statistics/test/itkImageToHistogramFilterTest2.cxx]: Optional parameters are usually in `[brackets]`, not `(parentheses)`.

Comment (blowekamp) [reply Modules/Numerics/Statistics/test/itkImageToHistogramFilterTest2.cxx]: done

File: Modules/Numerics/Statistics/include/itkImageToHistogramFilter.hxx [hunk 2]
```diff
@@ -98,10 +98,22 @@ ImageToHistogramFilter<TImage>::GetNumberOfInputRequestedRegions()
...
+  Superclass::StreamedGenerateData(inputRequestedRegionNumber);
+}
+
 
 template <typename TImage>
 void
-ImageToHistogramFilter<TImage>::BeforeStreamedGenerateData()
+ImageToHistogramFilter<TImage>::InitializeOutputHistogram()
 {
   Superclass::BeforeStreamedGenerateData();
```
Comment (blowekamp) [inline Modules/Numerics/Statistics/include/itkImageToHistogramFilter.hxx]: This should not be called here any more.

**Pull request merged**